langlib/langlib-test/src/test/resources/test-src/arraylib_test.bal

```diff
@@ -1552,27 +1922,67 @@ function testSort10() {
     };
 
     int[] sortedArr2 = arr.sort(array:DESCENDING, addFunc1);
+    int[] sortedTup2 = tup.sort(array:DESCENDING, addFunc1);
 
     assertValueEquality(sortedArr2.toString(), "[10,6,3,2,1,0]");
+    assertValueEquality(sortedTup2.toString(), "[10,6,3,2,1,0]");
 
     int[] sortedArr3 = array:sort(arr, array:ASCENDING, isolated function(int x) returns string[] => [x.toString(), "World"]);
+    int[] sortedTup3 = array:sort(tup, array:ASCENDING, isolated function(int x) returns string[] => [x.toString(), "World"]);
 
     assertValueEquality(sortedArr3.toString(), "[0,1,10,2,3,6]");
+    assertValueEquality(sortedTup3.toString(), "[0,1,10,2,3,6]");
 
     int[] sortedArr4 = arr.sort(array:DESCENDING, (i) => i.toString());
+    int[] sortedTup4 = tup.sort(array:DESCENDING, (i) => i.toString());
 
     assertValueEquality(sortedArr4, [6, 3, 2, 10, 1, 0]);
+    assertValueEquality(sortedTup4, [6, 3, 2, 10, 1, 0]);
 
     int?[] arr2 = [(), 1, 3, 10, 0, 6];
+    [int?...] tup2 = [(), 1, 3, 10, 0, 6];
 
     int?[] sortedArr5 = arr2.sort(array:DESCENDING, (i) => i);
+    int?[] sortedTup5 = tup2.sort(array:DESCENDING, (i) => i);
 
     assertValueEquality(sortedArr5, [10, 6, 3, 1, 0, ()]);
+    assertValueEquality(sortedTup5, [10, 6, 3, 1, 0, ()]);
 
     int[] arr3 = [];
+    [int...] tup3 = [];
     int[] sortedArr6 = arr3.sort(array:DESCENDING, (i) => i);
+    int[] sortedTup6 = tup3.sort(array:DESCENDING, (i) => i);
 
     assertValueEquality(sortedArr6, []);
+    assertValueEquality(sortedTup6, []);
+}
+
+type Grade2 1|15|"B";
+type Grade3 15|17|Grade3[];
+type Grade4 (Grade2|int)|Grade3;
+
+function testSortNegative() {
+    [int, float] tup1 = [5, 7.3];
```

Comment:
Do you have positive tests for such tuples with key functions? The scenarios I tried failed, due to existing issues, but I feel like those issues restrict how much testing we can do.

Reply:
Added in c5daaea654d92b306868ce09652d183a764da99f